### PR TITLE
Add .editorconfig

### LINKED ISSUE #54: 🚀💻 – .editorconfig?
**Describe the feature you want and how it meets your needs or solves a problem**

As a developer and reviewer I want to have consistent formatting of code across all contributions. [.editorconfig](https://editorconfig.org/) files can configure a wide variety of text editors to function similarly, e.g., trim trailing whitespace, set tabs to 4 spaces for python files, but 2 spaces for yaml, &c.

**Describe the solution you'd like**

I'd like a project editor config file, happy to contribute my own.

**Describe alternatives you've considered**

We could add standard editor settings to the contributing document, but those would not automatically apply.

**Additional context**

### PR DESCRIPTION
**Pull request by botanize**
Uses typical defaults and de facto standards already in place in this repo.

Closes #54

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+insert_final_newline = true
+
+[*.py]
+indent_size = 4
+
+[{.git*,Makefile}]
+indent_style = tab
+
+[{COMMIT_EDITMSG,MERGE_MSG,SQUASH_MSG}]
+max_line_length = 72
```
Comment (e-lo) [inline .editorconfig:19]: I'd prefer if we use 99. (I know we don't have that yet other places). Thoughts? It seems much more do-able and makes it so you can fit a lot more into in-line comments without having to break them up.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# double whitespace at end of line
+# denotes a line break in Markdown
+trim_trailing_whitespace = false
+x-soft-wrap-text = true
+
+[*.json]
+insert_final_newline = ignore
+
+[*{.ini,.flake8}]
+indent_size = 4
```
Comment (e-lo) [inline .editorconfig:31]: Wondering why this isn't 2? (real question...I'm not sure the diff on this vs above)

**Review by e-lo — CHANGES_REQUESTED** [review]
A few minor questions/requests/thoughts.

File: .editorconfig (hunk 1 above)
Comment (botanize) [reply .editorconfig:19]: This is just for commit messages. I didn't set a max line length otherwise.

File: .editorconfig (hunk 2 above)
Comment (botanize) [reply .editorconfig:31]: Only reason I chose 4 here was that was what was used in the existing files in the repo. I can remove it, and ini/flake8 files will inherit from [*] above (indent_size = 2).

File: .editorconfig (hunk 1 above)
Comment (e-lo) [reply .editorconfig:19]: clearly didn't read this close enough

**Review by e-lo — APPROVED** [review]
LGTM

**Pull request merged**